File: playground-dart/benchmark/n206/reverse_list_benchmark.dart
import 'package:benchmark_harness/benchmark_harness.dart';
import 'package:playground_dart/datastructure/list_node.dart';
import 'package:playground_dart/n206/solution.dart';

class ReverseListBenchmark extends BenchmarkBase {
  ReverseListBenchmark() : super('ReverseList');

  final solution = Solution();
  ListNode? head;

  @override
  void setup() {
    head = ListNode.fromList([1, 2, 3, 4, 5]);
  }

  @override
  void run() {
    solution.reverseList(head);
  }
}

void main() {
  ReverseListBenchmark().report();
}

File: playground-kotlin/src/jmh/kotlin/com/releetcode/n206/ReverseListBenchmark.kt
package com.releetcode.n206

import com.releetcode.datastructure.ListNode
import org.openjdk.jmh.annotations.*
import java.util.concurrent.TimeUnit

@State(Scope.Benchmark)
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.NANOSECONDS)
open class ReverseListBenchmark {
    private val solution = Solution()
    private var head: ListNode? = null

    @Setup
    fun setup() {
        head = ListNode.fromArray(intArrayOf(1, 2, 3, 4, 5))
    }

    @Benchmark
    fun benchmarkReverseList(): ListNode? {
        return solution.reverseList(head)
    }
}
